api/v1alpha8/openstackcluster_webhook.go

```diff
@@ -116,6 +116,13 @@ func (r *OpenStackCluster) ValidateUpdate(oldRaw runtime.Object) (admission.Warn
 		r.Spec.APIServerPort = 0
 	}
 
+	// Allow to remove the bastion spec only if it was disabled before.
+	if r.Spec.Bastion == nil {
+		if old.Spec.Bastion != nil && old.Spec.Bastion.Enabled {
```

Comment:
note that I didn't add a guard to compare the old instance with the new one because there are complicated objects with filters. If we want to go down that path, let me know although I'm not sure it's desired.

Otherwise let me know if the doc content is enough.